packages/block-editor/src/utils/transform-styles/index.js

```diff
@@ -1,36 +1,36 @@
 /**
- * WordPress dependencies
+ * External dependencies
  */
-import { compose } from '@wordpress/compose';
-
-/**
- * Internal dependencies
- */
-import traverse from './traverse';
-import urlRewrite from './transforms/url-rewrite';
-import wrap from './transforms/wrap';
+import postcss from 'postcss';
+import wrap from 'postcss-prefixwrap';
+import rebaseUrl from 'postcss-urlrebase';
 
 /**
  * Applies a series of CSS rule transforms to wrap selectors inside a given class and/or rewrite URLs depending on the parameters passed.
  *
- * @param {Object|Array} styles           CSS rules.
- * @param {string}       wrapperClassName Wrapper Class Name.
+ * @typedef {Object} EditorStyle
+ * @property {string}        css              the CSS block(s), as a single string.
+ * @property {?string}       baseURL          the base URL to be used as the reference when rewritting urls.
+ * @property {?string[]}     ignoredSelectors the selectors not to wrap.
+ *
+ * @param    {EditorStyle[]} styles           CSS rules.
+ * @param    {string}        wrapperSelector  Wrapper selector.
  * @return {Array} converted rules.
  */
-const transformStyles = ( styles, wrapperClassName = '' ) => {
-	return Object.values( styles ?? [] ).map( ( { css, baseURL } ) => {
```

Comment:
We don't need to do that if we ensure that we're always getting an array. I checked the usages and the only place where this might not happen is https://github.com/WordPress/gutenberg/pull/49521/files#diff-8ee3a55dcb37ec88dd2c1e1c752f2996038b5ee8dec3b73a44c9d4f39ecc2a75, so I updated it.

Reply:
I added a task in the PR test instructions: "Add the HTML Preview block and check that adding custom styles do not break it."